apps/api/src/app/workflows-v2/util/utils.ts

```diff
@@ -241,7 +241,7 @@ export function mergeCommonObjectKeys(
           sourceValue as Record<string, unknown>
         );
       } else {
-        merged[key] = sourceValue ?? targetValue;
+        merged[key] = key in source ? sourceValue : targetValue;
```

Comment:
Check presence of key in payload instead of a value - If I put `null` in preview payload as a value of some attribute (e.g. to check `ifShowKey` condition), I want it to not get it overwritten by placeholder string